server/route.go

```diff
@@ -491,6 +496,7 @@ func (c *client) processRouteInfo(info *Info) {
 
 	supportsHeaders := c.srv.supportsHeaders()
 	clusterName := c.srv.ClusterName()
+	isSusceptible := c.srv.isClusterNameSusceptibleToChanges()
```

Comment:
not sure about the chosen name here... a bit longer, but maybe more telling: `clusterNameChangeAllowed`?